src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -250,12 +256,21 @@ internal static async Task HandleExceptionAsync(TimeoutAttribute timeout, Except
                     // TODO: Refactor all of this logging to be implemented as a separate ILogger.
                     FunctionOutputLogger.SetOutput(outputLog);
 
+                    // Getting retry count from TriggerDetails
+                    int retryCount = 0;
```

Comment:
Not sure I understand what is going on here.

Reply:
`TriggerDetails` is used to pass current retry count to `ExecutionContext`

https://github.com/Azure/azure-webjobs-sdk/pull/2463/files#diff-d4edc2f1f55dd03bebe0b9fa640a1656R407

https://github.com/Azure/azure-functions-eventhubs-extension/pull/56/files#diff-5e1bd09eff27cb8f09152894835d29b6R88

Reply:
Since the executor fully controls the retries, I don't see why you need to be fishing this out of TriggerDetails like this. Instead, you should just pass the retry count in from wherever it is being performed upstream

Reply:
Udpated!